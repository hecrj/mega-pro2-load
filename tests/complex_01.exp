1 2 3 8 0 0 0 4 0 0 5 6 0 0 7 9 0 0 0

       1
     /    \
    2      5
   / \    / \
  3   4  6   7
 /          /
8          9

Tamanys películes:
1 -> 20
2 -> 100

Tots el servidors tenen la mateixa velocitat (10) de transferència.

- Primera petició:
Es demana la película 1, disponible a 2, 6, 7, 8 i 9.
Els camins possibles son (amb la profunditat):
2    (2)
2-8  (4)
6    (3)
7    (3)
7-9  (4)
Els camins 2-8 i 7-9 están empatats en velocitat (unitaris) i profunditat, cal escollir el de
l'esquerra: 2-8.

- Consulta servidors:
Com que la consulta es fa 1 ut després de la primera petició no hi ha servidors ocupats.

- Segona petició:
Es demana la película 2, disponible a 1, 3, 4 i 5.
Cal retornar al camí amb velocitat màxima, doncs no es pot servir en 1 ut.
Els camins possibles son (amb la profunditat):
1
1-3 (3)
1-4 (3)
1-5 (2)
Cal escollir el camí 1-5, doncs es troba empatat amb 1-3 i 1-4 amb velocitat però es
menyos profund.

- Consulta peticions:
La consulta es produeix 1 ut després de la segona petició. Com que la petició no es
podia servir en 1 ut cal que es mostri la seva informació. Com que la película té
un tamany de 100 i la velocitat del camí escollit es 20, la petició trigarà 5 uts
en finalitzar.

- Consulta servidors:
La consulta es produeix 2 uts després de la segona petició. Per tant, encara queden 3
uts per a que la petició acabi. Cal mostrar els servidors 1-5 com ocupats amb un temps
per finalitzar de 3 uts.

- Tercera petició:
Es produeix 3 uts després de la segona petició i es demana la película 2. Els servidor 1-5
segueixen ocupats, per tant, ara tenen disponible la película: 3 i 4.
Ambdós camins estan empatats en velocitat i profunditat, cal escollir de l'esquerra, 3.
La petició trigarà 100 / 10 = 10 uts.

- Cuarta petició:
Es produeix 1 ut després de la tercera petició, 4 després de la segona, i es demana la
película 2. L'únic servidor disponible, per tant, es el 4.
La petició trigarà 100 / 10 = 10 uts.

- Cinquena petició:
Es produeix 5 uts després de la segona petició, per tant, ja ha acabat. Es demana la
película 2. Els servidors 1 i 5 estàn disponibles de nou i es tornen a seleccionar,
doncs són els únics disponibles.

- Sisena petició:
Es produeix 1 ut després de la cinquena petició i es demana la película 2 de nou.
Ara, però, tots els servidors que tenen la película 2 estàn ocupats per les tres anteriors
peticions (només han passat 3 uts de la tercera petició). Per tant, la petició no es pot servir.

- Consulta servidors:
Es produeix 1 ut després de la sisena petició. Cal que es mostrin tots el servidors que tenen
la película 2: 1, 3, 4 i 5 amb els identificadors i temps restants pertinents.

- Consulta de película 1:
Es fa una consulta entre els temps 0 i 9. Això comprèn totes les peticions realitzades, per tant,
la película més descarregada es la 2 amb un total de 4 vegades (la que no s'ha pogut servir no es
comptabilitza).

- Consulta de película 2:
Es fa una consulta entre els temps 0 i 1. Només comprén la primera petició, per tant, la película
descarregada es la 1 i un únic cop.

- Consulta de película 3:
Es fa una consulta entre els temps 0 i 2. Comprén la primera y segona petició, la película 1 i 2 estàn
empatades en quant a nombre de descàrregues. Cal mostrar la película 1, doncs, té un identificador menor.

- Consulta de película 4:
Es fa una consulta entre els temps 0 i 5. Comprén la primera, segona i tercera peticions. La película 2 amb
2 descàrregues guanya a la película 1 amb 1 descàrrega.

- Consulta de película 5:
Es fa una consulta entre els temps 0 i 200. 200 es un temps gran que podria causar errors en implementacions
incorrectes (per violació . El resultat ha de ser el mateix que a la consulta de película 1.

- Finalització del programa:
El programa ha de finalitzar correctament.

Entrada sense comentar
===========================
2
20 100

9
1 2 3 8 0 0 0 4 0 0 5 6 0 0 7 9 0 0 0

10
1 2

10
1 1

10
1 2

10
1 2

10
1 2

10
1 1

10
1 1

10
1 1

10
1 1


-1 1 0
-2 1

-1 2 2
-2 3

-3 4

-1 2 5
-1 2 6
-1 2 7
-1 2 8

-3 9

-5 0 9
-5 0 1
-5 0 2
-5 0 5
-5 0 200

-6
=========================

Sortida sense comentar
=========================
Peticion procesada y servidores
0 1
2 8
Peticiones pendientes
0
Peticion procesada y servidores
1 5
1 5
Peticiones pendientes
1 2 2 7
Servidores ocupados
1 1 3
5 1 3
Peticion procesada y servidores
2 10
3
Peticion procesada y servidores
3 10
4
Peticion procesada y servidores
4 5
1 5
Peticion procesada y servidores
5 0
Servidores ocupados
1 4 3
3 2 6
4 3 7
5 4 3
Pelicula mas solicitada
2 4
Pelicula mas solicitada
1 1
Pelicula mas solicitada
1 1
Pelicula mas solicitada
2 2
Pelicula mas solicitada
2 4
===========================