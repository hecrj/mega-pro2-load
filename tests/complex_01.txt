2
20 100

9
1 2 3 8 0 0 0 4 0 0 5 6 0 0 7 9 0 0 0

10
1 2

10
1 1

10
1 2

10
1 2

10
1 2

10
1 1

10
1 1

10
1 1

10
1 1


-1 1 0
-2 1

-1 2 2
-2 3

-3 4

-1 2 5
-1 2 6
-1 2 7
-1 2 8

-3 9

-5 0 9
-5 0 1
-5 0 2
-5 0 5

-6
