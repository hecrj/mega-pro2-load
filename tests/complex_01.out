Peticion procesada y servidores
0 1
2 8
Peticiones pendientes
0
Peticion procesada y servidores
1 5
1 5
Peticiones pendientes
1 2 2 7
Servidores ocupados
1 1 3
5 1 3
Peticion procesada y servidores
2 10
3
Peticion procesada y servidores
3 10
4
Peticion procesada y servidores
4 5
1 5
Peticion procesada y servidores
5 0
Servidores ocupados
1 4 3
3 2 6
4 3 7
5 4 3
Pelicula mas solicitada
2 4
Pelicula mas solicitada
1 1
Pelicula mas solicitada
1 1
Pelicula mas solicitada
2 2
Pelicula mas solicitada
2 4
